parquetio/src/main/java/com/adobe/platform/ecosystem/examples/parquet/write/ParquetIOWriterImpl.java

```diff
@@ -93,23 +102,67 @@ private GroupType getGroupTypeField(ParquetIOField field) {
         GroupBuilder<GroupType> groupTypeBuilder;
         if(field.isRepetetive()) {
             groupTypeBuilder = Types.repeatedGroup();
+        } else if(field.isRequired()) {
+            groupTypeBuilder = Types.requiredGroup();
         } else {
             groupTypeBuilder = Types.optionalGroup();
         }
 
         // Iterate on children and recursively populate groups.
         if(field.getSubFields() != null) {
             for(ParquetIOField subField : field.getSubFields()) {
-                if(subField.isPrimitive()) {
-                    groupTypeBuilder.addField(getPrimitiveTypeField(subField));
-                } else {
-                    groupTypeBuilder.addField(getGroupTypeField(subField));
-                }
+                groupTypeBuilder.addField(
+                    getTypeFromField(subField)
+                );
             }
         }
         return groupTypeBuilder.named(field.getName());
     }
 
+    private Type getListType(ParquetIOField field) {
+        ListBuilder<GroupType> listBuilder;
+        if (field.getRepetitionType() == ParquetIORepetitionType.REQUIRED) {
+            listBuilder = Types.requiredList();
+        } else {
+            listBuilder = Types.optionalList();
+        }
+
+        // Assumption here is that length of
+        // field.getSubFields() will be 1.
```

Comment:
We already have cases in XDM where list contain more than sub fields>1, please refer to below sinppet of EE XDM.

```
        "segmentMemberships": {
            "title": "Segment Memberships",
            "description": "The segments associated with this experience event",
            "type": "array",
            "meta:xdmType": "array",
            "items": {
                "type": "object",
                "properties": {
                    "**segmentID**": {
                        "title": "Segment ID",
                        "description": "The identity of the segment or snapshot definition in with the domain of the specific system that processes that type of segment.",
                        "meta:xdmField": "xdm:segmentID",
                        "type": "object",
                        "properties": {
                            "_id": {
                                "title": "Identifier",
                                "type": "string",
                                "format": "uri-reference",
                                "description": "Identity of the Segment in the related namespace.",
                                "meta:xdmType": "string",
                                "meta:xdmField": "@id"
                            },
                            "namespace": {
                                "title": "Namespace",
                                "description": "The namespace associated with the `xid` attribute.",
                                "meta:xdmField": "xdm:namespace",
                                "type": "object",
                                "properties": {
                                    "code": {
                                        "title": "Code",
                                        "type": "string",
                                        "description": "The code is a human readable identifier for the namespace and can be used to request the technical namespace id which is used for identity graph processing.",
                                        "meta:xdmType": "string",
                                        "meta:xdmField": "xdm:code"
                                    }
                                },
                                "required": [
                                    "code"
                                ],
                                "xdmVersion": "0.9.9.7",
                                "meta:altId": "_xdm.context.namespace",
                                "meta:xdmType": "object",
                                "meta:status": "stabilizing",
                                "$schema": "http://json-schema.org/draft-06/schema#",
                                "$id": "context/namespace",
                                "_refId": "context/namespace"
                            }
                        },
                        "meta:xdmType": "object"
                    },
                    "**status**": {
                        "title": "Status",
                        "type": "string",
                        "description": "Is the segment participation realized as part of the current request.",
                        "default": "realized",
                        "enum": [
                            "existing",
                            "realized",
                            "exited"
                        ],
                        "meta:enum": {
                            "existing": "Entity continues to be in the segment",
                            "realized": "Entity is entering the segment",
                            "exited": "Entity is exiting the segment"
                        },
                        "meta:xdmType": "string",
                        "meta:xdmField": "xdm:status"
                    }
                },
                "meta:xdmType": "object"
            },
            "meta:xdmField": "xdm:segmentMemberships"
        }
```